packages/swarm-components/src/FieldLabel.test.jsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+
+import { toMatchImageSnapshot } from 'jest-image-snapshot';
+expect.extend({ toMatchImageSnapshot });
+
+import getScreenRenderer from './testUtils/screenRenderer';
+
+import { getTestMarkup } from './testUtils/testUtils.js';
+
+import FieldLabel from './FieldLabel';
+import Checkbox from './Checkbox';
+
+describe('FieldLabel', () => {
+	let renderer;
+
+	// This is ran when the suite starts up.
+	beforeAll(async () => {
+		renderer = await getScreenRenderer({
+			viewport: { width: 250, height: 200 },
+			// verbose: true,
+		});
+	});
+
+	// This is ran when the suite is done - stop the renderer.
+	afterAll(() => {
+		// comment next line out if you want to open it in your browser for debugging
+		return renderer.stop();
+	});
+
+	const testCases = [
+		['Label alone', <FieldLabel key="alone">Field Label</FieldLabel>],
+		[
+			'Pass focus',
+			<div key="focus">
+				<FieldLabel id="swarm-external-label" htmlFor="swarm-checkbox">
+					External checkbox Label
+				</FieldLabel>
+				<Checkbox id="swarm-checkbox">Checkbox&apos;s own text</Checkbox>
```

Comment:
Why does `FieldLabel` have a `pass focus` state? From my understanding `FieldLabel` should be completely independent of the form element. Is that not the case?

Reply:
It's HTML core functionality of labels (not React state) where click on the label should focus the element specified in React's `htmlFor` / HTML's `for` attribute.

Reply:
So it is a functional test (not necessarily purely visual) that ensures that focus is passed (see blue border around checkbox in a screenshot).